### DISCUSSION TITLE
mon restarts frequently

### DISCUSSION THREAD

**Discussion by zhangdeshuai1999** [rook/rook#13950] [Q&A]
I deployed the latest rook-ceph in k8s.1.23 version, and there were no faults or problems during the deployment. However, after the deployment was completed, I used tools to enter the discovery timeout, and then I went to check my mon, and found that my mon and mgr kept restarting, but I could not find the problem through mon log.

![WechatIMG239](https://github.com/rook/rook/assets/141296859/d8a980e3-2727-4897-bd82-af94b0723854)
![WechatIMG240](https://github.com/rook/rook/assets/141296859/77f1b7ac-1196-4bad-a35f-8c64cb0f4ef0)
![WechatIMG241](https://github.com/rook/rook/assets/141296859/ab258019-89a8-4295-9eb9-47c2ca4744cb)
![WechatIMG242](https://github.com/rook/rook/assets/141296859/080b8c07-76fa-4099-8228-d5172d8e2f0e)

**Comment by parth-gr** [comment]
Looks like there a network issue, so the liveness probe is failing and not able to run the admin-socket command,

**Reply by zhangdeshuai1999** [reply]
I checked the network. It's fine

**Reply by parth-gr** [reply]
can you run the liveness probe cmd `ceph --admin-daemon /run/ceph/ceph-mon.asok mon_status`
?

**Reply by zhangdeshuai1999** [reply]
I don't understand

**Reply by parth-gr** [reply]
can you run the toolbox pod and sh into it and try to run the above cmd.

**Reply by zhangdeshuai1999** [reply]
yes

**Reply by travisn** [reply]
That liveness probe is run on the mon daemon pod itself, so you'll want to try that command on the mon pod, not the toolbox pod. 

Since the livenesss probe is failing, there must be some issue within the pod for the mon to connect to its admin port.

**Reply by zhangdeshuai1999** [reply]
[mon.txt](https://github.com/rook/rook/files/14644316/mon.txt)
This is what I saw in mon-c, please help me have a look, thank you

**Reply by zhangdeshuai1999** [reply]
The other two mon, pod state is never ready, exec -it can't execute cmd

**Reply by zhangdeshuai1999** [reply]
I redeployed ceph. I went into the cmd of 3 mons and checked ceph-mon.asok mon_status. Their Mons are the same, but the IP addresses of extra_probe_peers seem to be different.
[mona.txt](https://github.com/rook/rook/files/14644774/mona.txt)
[monb.txt](https://github.com/rook/rook/files/14644776/monb.txt)
[monc.txt](https://github.com/rook/rook/files/14644777/monc.txt)

**Reply by microyahoo** [reply]
Does your IP address change frequently? I see that your newly deployed IPs are different from the previous one.

**Reply by zhangdeshuai1999** [reply]
My IP address is automatically assigned by the container group IP address pool

**Reply by microyahoo** [reply]
do you mean restarting mon pod will change ip address? Why not use hostNetwork?

**Reply by zhangdeshuai1999** [reply]
Is the network defined in CRD？
Such as the following
network:
    hostNetwork: true

**Reply by zhangdeshuai1999** [reply]
Now is the problem the network?

**Reply by microyahoo** [reply]
https://rook.io/docs/rook/v1.12/Storage-Configuration/Advanced/ceph-configuration/#use-hostnetwork-in-the-cluster-configuration

**Reply by zhangdeshuai1999** [reply]
<img width="861" alt="图片" src="https://github.com/rook/rook/assets/141296859/8d942092-46a0-4a10-a786-f50208c31649">
<img width="651" alt="图片" src="https://github.com/rook/rook/assets/141296859/b320e0f6-6514-49cc-aa8b-f4252632ac2d">
The host network still appeared, and the redeployment was normal at first, but quickly became problematic. It's so weird.

It is very strange that the file format of disk is cephb, and the file format of one is cephbluestore、The normal mon runs on the cephblue node, and the other two mon restarts frequently run on ceph-b.
<img width="321" alt="图片" src="https://github.com/rook/rook/assets/141296859/704e7148-775e-4f93-9c87-846ef9dd4938">
<img width="167" alt="图片" src="https://github.com/rook/rook/assets/141296859/0fdb1e94-2557-4c8f-b07e-fe5cf58c3ae1">

**Reply by microyahoo** [reply]
I'm curious what's ceph-b? mon data reside at root disk, has nothing to do with ceph_bluestore disk.

**Reply by zhangdeshuai1999** [reply]
Yes. I know that mon's data is under /var/lib/rook/. In this case, I suspect there may be a problem with the mainframe.

**Reply by zhangdeshuai1999** [reply]
All three of my ceph nodes are masters, does it have anything to do with this

**Reply by subhamkrai** [reply]
@zhangdeshuai1999 did you clean `/var/lib/rook` before redployment?

**Reply by zhangdeshuai1999** [reply]
The network between mon, which had been removed and redeployed after /var/lib/rook, was fine, but the probe timed out and I now know it's probably a kernel issue, so I'm ready to upgrade the kernel and try.

**Reply by subhamkrai** [reply]
also check this https://rook.github.io/docs/rook/latest/Troubleshooting/ceph-common-issues/#operator-fails-to-connect-to-the-mon

**Reply by zhangdeshuai1999** [reply]
Yeah, I checked. It's fine
